File: Android/unity3d/src/main/java/io/loomx/unity3d/LoginCallback.java
package io.loomx.unity3d;

import android.app.Dialog;
import android.support.annotation.NonNull;

import com.auth0.android.authentication.AuthenticationException;
import com.auth0.android.result.Credentials;

/**
 * Callback called on success/failure of an Identity Provider authentication.
 * Only one of the success or failure methods will be called for a single authentication request.
 */
public interface LoginCallback {
    /**
     * Called with an AuthenticationException that describes the error.
     *
     * @param exception cause of the error
     */
    void onFailure(String exception);

    /**
     * Called when the authentication is successful using web authentication against Auth0
     *
     * @param accessToken Auth0 access token
     */
    void onSuccess(@NonNull String accessToken);
}

File: Assets/LoomSDK/Android/LoginCallback.cs
using System;
using UnityEngine;

namespace Loom.Unity3d.Android
{
    class LoginCallback : AndroidJavaProxy
    {
        public Action<string> OnSuccess;
        public Action<string> OnFailure;

        public LoginCallback() : base("io.loomx.unity3d.LoginCallback")
        {
        }

        public void onFailure(string exception)
        {
            this.OnFailure(exception);
        }

        public void onSuccess(string accessToken)
        {
            this.OnSuccess(accessToken);
        }
    }
}